documentation/projects/proposals/document_all_media_properties/20240313-implementation_plan_document_media_properties_api.md

```diff
@@ -0,0 +1,239 @@
+# 2024-03-25 Implementation Plan: Document all media properties in the API
+
+**Author**: @dhruvkb
+
+## Reviewers
+
+<!-- Choose two people at your discretion who make sense to review this based on their existing expertise. Check in to make sure folks aren't currently reviewing more than one other proposal or RFC. -->
+
+- [ ] @obulat
+- [x] @krysal
+
+## Project links
+
+<!-- Enumerate any references to other documents/pages, including milestones and other plans -->
+
+- [Project Thread](https://github.com/WordPress/openverse/issues/412)
+- [Project Proposal](/projects/proposals/document_all_media_properties/20230307-project_proposal.md)
+
+## Overview
+
+This implementation plan devises a method for documenting the media properties
+for media models defined in the API.
+
+## Expected Outcomes
+
+The outcome of this implementation plan will be a documentation page for the
+media properties pertaining to the models in the Django API, similar to the
+documentation for the
+[media properties for the catalog](/projects/proposals/document_all_media_properties/20230426-implementation_plan_documenting_media_properties.md).
+
+This implementation plan will make changes to the location of the catalog media
+properties documentation so that it can be extended to include information about
+the API and, in the future, the frontend.
+
+## Step-by-step plan
+
+The plan consists of the following steps:
+
+- Define management command and `just` recipe.
+- Introspect Django models.
+- Merge with additional notes.
+- Generate documentation page.
+- Ensure freshness using CI.
+
+## Step details
+
+### Define management command and `just` recipe
+
+The complete process of generating the documentation will be encapsulated in a
+Django management command. This command will be named `documentmedia` and will
+be defined in the `api` app in the API project.
+
+On its own the command will be insufficient to write the documentation page to
+the documentation project as it will run inside the Django context and will not
+have access to the filesystem. This management command will write the
+documentation to the API project root.
+
+We will invoke this command from a `just` recipe which will move the file to its
+destination, lint it and also function as the CI test which returns a non-zero
+exit code when the file is modified in the process. This recipe will be named
+`gen-media-props`.
```

Comment:
We could name the script more generally, like `generate-docs` with argument of `media-props` (as in the catalog PR). If we ever have other generated docs in the API, then it would be easier to update.
Just a suggestion to unify the naming across the stack.

Reply:
That makes sense, I didn't consider the convention that was established in the catalog by #3620. Updated in 166bb42.